tools/astarte_export/lib/export.ex

```diff
@@ -0,0 +1,282 @@
+
+  #######################################################################
+  # This function will seralize the realm data  to the below format 
+  # {:tag_name, %{}, ""} 
+  # which is used by XmlBuilder library to convert to a xml string 
+  #######################################################################
+
+  @spec seralize_xml(atom(), struct() | keyword()) :: {atom(), map(), String.t() | keyword()}
+
+  defp seralize_xml(tag, options) do
+    {tag, get_attributes(tag, options), get_value(tag, options)}
+  end
+
+  ######################################################################
+  #
+  # get_attributes/2 return a map which hold all attributes to be 
+  # included in a specific XML tag
+  #
+  ######################################################################
+
+  @spec get_attributes(atom(), struct() | keyword()) :: map()
+
+  defp get_attributes(:device, options) do
+    state = options[:state]
+    %{:device_id => state.device_id}
+  end
+
+  defp get_attributes(:protocol, options) do
+    state = options[:state]
+    %{:revision => state.revision, :pending_empty_cache => state.pending_empty_cache}
+  end
+
+  defp get_attributes(:registration, options) do
+    state = options[:state]
+
+    %{
+      :secret_bcrypt_hash => state.secret_bcrypt_hash,
+      :first_registration => state.first_registration
+    }
+  end
+
+  defp get_attributes(:credentials, options) do
+    state = options[:state]
+
+    %{
+      :inhibit_request => state.inhibit_request,
+      :cert_serial => state.cert_serial,
+      :cert_aki => state.cert_aki,
+      :first_credentials_request => state.first_credentials_request
+    }
+  end
+
+  defp get_attributes(:stats, options) do
+    state = options[:state]
+
+    %{
+      :total_received_msgs => state.total_received_msgs,
+      :total_received_bytes => state.total_received_bytes,
+      :last_connection => state.last_connection,
+      :last_disconnection => state.last_disconnection,
+      :last_seen_ip => state.last_seen_ip
+    }
+  end
+
+  defp get_attributes(:interfaces, state) do
+    %{}
+  end
+
+  defp get_attributes(:interface, state) do
+    %{
+      :name => state.interface_name,
+      :major_version => state.major_version,
+      :minor_version => state.minor_version,
+      :active => state.active
+    }
+  end
+
+  defp get_attributes(:datastream, {_type, state}) do
+    %{:path => state.path}
+  end
+
+  defp get_attributes(:property, state) do
+    %{:path => state[:path], :reception_timestamp => state[:reception_timestamp]}
+  end
+
+  defp get_attributes(:object, state) do
+    %{:reception_timestamp => state[:reception_timestamp]}
+  end
+
+  defp get_attributes(:value, state) do
+    %{:reception_timestamp => state[:reception_timestamp]}
+  end
+
+  defp get_attributes(:item, value) do
+    %{:name => value[:v_realpathdatavalue]}
+  end
+
+  defp get_attributes(_tag, _value) do
+    %{}
+  end
+
+  ###############################################################
+  #
```

Comment:
I think we should use elixir style documentation

Reply:
removed those comments